libs/s25main/GamePlayer.cpp

```diff
@@ -376,6 +377,19 @@ void GamePlayer::RemoveBuildingSite(noBuildingSite* bldSite)
     buildings.Remove(bldSite);
 }
 
+bool GamePlayer::IsHQTent() const
+{
+    if(const nobHQ* hq = GetHQ())
+        return hq->IsTent();
+    return false;
+}
+
+void GamePlayer::SetHQIsTent(bool isTent)
```

Comment:
This and `GetHQ` is basically the code moved from `ModifyHQ`. What was the motivation?

I'm thinking about the case where there are multiple HQs (e.g. the S2 cheats had it) and one might want to specify which HQ to change. Putting the code in the caller uses the world to get a specific HQ and changes that.

Although we already have `GetHQPos` which returns the position of "a" HQ, IIRC the last one added that still exists.